### DISCUSSION TITLE
HA solution for infinity server?

### DISCUSSION THREAD

**Discussion by bash99** [infiniflow/infinity#1578] [General]
If I want to deploy infinity server to production, how do I achieving basic high availability?

I don't found any section about this in document.

**Comment by JinHai-CN** [comment]
We plan to integrate S3 with Infinity in next release, which will provide the HA of the data storage.

**Reply by bash99** [reply]
So it'll operated like a share storage architecture?

     /----- Infinity Server ----\
LB ------ Infinity Server -----  S3 Storage?
     \----- Infinity Server ----/
     
How all Infinity Server get notified when S3 Storage is updated?

**Reply by JinHai-CN** [reply]
Yes, the shared storage architecture.

What do you mean the update of S3 storage? Updating the S3 server version, increasing the storage capacity?

**Reply by bash99** [reply]
> Yes, the shared storage architecture.
> 
> What do you mean the update of S3 storage? Updating the S3 server version, increasing the storage capacity?

I mean when add new data to database, one of the Infinity Server will write the update to S3 Storage, how another Infinity Server got the update? (I guess the Infinity Server will cache most index in memory?)

**Reply by JinHai-CN** [reply]
Suppose server A updates the data and the query occurs on server B. If we need the updated data, B server will send the query to A server for the log tail data that has just been updated. Of course, other modes can be selected here as well, reducing real-time requirements and improving performance.
